### PR TITLE
NAS-110543 / 21.08 / Added new field 'name_regex'

### PR DESCRIPTION
**Pull request by RehanY147**
NAS-110543

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
The following comments from previous pull request are still valid:
1. If I use Regular Expression, Radio button is not set when I edit replication task.
2. Would it be possible to talk to Vladimir and come up with better name for Naming Schema or Snapshot Name Regular Expression label. Something like Filter snapshots?

Otherwise seems to be working okay.

File: src/app/pages/data-protection/replication/replication-form/replication-form.component.ts [hunk 1]
```diff
@@ -1550,4 +1559,28 @@ export class ReplicationFormComponent implements FormConfiguration {
   isCustActionVisible(actionId: string): boolean {
     return actionId === 'wizard_add' && this.pk === undefined;
   }
+
+  toggleNamingSchemaOrRegex(): void {
+    const directionValue = this.entityForm.formGroup.controls['direction'].value;
+    const disabled = true;
+    const enabled = false;
```
[the remark this answers is not in the document]
Comment (RehanY147) [reply src/app/pages/data-protection/replication/replication-form/replication-form.component.ts:1566]: Alright. Just seemed confusing to me to always remember that `true` means `disabled`. So I made variables.

[the remark this answers is not in the document]
Comment (themylogin) [reply src/app/pages/data-protection/replication/replication-form/replication-form.component.ts:1566]: For push replication:
```
Also include snapshots with the name:
[*] Matching naming schema
[ ] Matching regular expression
```
For pull replication:
```
Also include snapshots with the name:
[*] Matching naming schema
[ ] Matching regular expression
```

**Review by undsoft — APPROVED** [review]
Minor text change from Vladimir, otherwise UI is working correctly.

**Pull request merged**